apps/sms/views/shared/test/unit/contacts_test.js

```diff
@@ -262,169 +121,130 @@ suite('Contacts', function(done) {
-
-        // navigator.mozContacts.find was called?
-        assert.equal(mHistory.length, 1);
       }).then(done, done);
     });
 
 
     test('predominate first, lower', function(done) {
-      var mozContacts = navigator.mozContacts;
+      var result = MockContact.list();
 
-      Contacts.findContactByString('pepi o').then(function(contacts) {
-        var mHistory = mozContacts.mHistory;
+      navigator.mozContacts.find.returns(Promise.resolve(result));
 
+      Contacts.findContactByString('pepi o').then(function(contacts) {
         // No contacts were found
         assert.ok(Array.isArray(contacts));
         assert.equal(contacts.length, 1);
-
-        // navigator.mozContacts.find was called?
-        assert.equal(mHistory.length, 1);
       }).then(done, done);
     });
 
     test('predominate last, lower', function(done) {
-      var mozContacts = navigator.mozContacts;
+      var result = MockContact.list();
 
-      Contacts.findContactByString('o pepi').then(function(contacts) {
-        var mHistory = mozContacts.mHistory;
+      navigator.mozContacts.find.returns(Promise.resolve(result));
 
+      Contacts.findContactByString('o pepi').then(function(contacts) {
         // No contacts were found
         assert.ok(Array.isArray(contacts));
         assert.equal(contacts.length, 1);
-
-        // navigator.mozContacts.find was called?
-        assert.equal(mHistory.length, 1);
       }).then(done, done);
     });
 
     test('multi-word name', function(done) {
-      var mozContacts = navigator.mozContacts;
+      var result = MockContact.list([
+        { givenName: ['Mary Anne'], familyName: ['Jones'] }
+        ]);
 
-      Contacts.findContactByString('mary anne').then(function(contacts) {
-        var mHistory = mozContacts.mHistory;
+      navigator.mozContacts.find.returns(Promise.resolve(result));
 
+      Contacts.findContactByString('mary anne').then(function(contacts) {
         // No contacts were found
         assert.ok(Array.isArray(contacts));
         assert.equal(contacts.length, 1);
-
-        // navigator.mozContacts.find was called?
-        assert.equal(mHistory.length, 1);
       }).then(done, done);
     });
 
     test('name first, part of tel number last', function(done) {
-      var mozContacts = navigator.mozContacts;
+      var result = MockContact.list();
 
-      Contacts.findContactByString('Pepito 8888').then(function(contacts) {
-        var mHistory = mozContacts.mHistory;
+      navigator.mozContacts.find.returns(Promise.resolve(result));
 
+      Contacts.findContactByString('Pepito 8888').then(function(contacts) {
         // contacts were found
         assert.ok(Array.isArray(contacts));
         assert.equal(contacts.length, 1);
-
-        // navigator.mozContacts.find was called?
-        assert.equal(mHistory.length, 1);
       }).then(done, done);
     });
 
     test('part of tel number first, name last', function(done) {
-      var mozContacts = navigator.mozContacts;
```

Comment:
Yes! because i used stub.restore() in suiteTeardown(), it restores the API after all the tests and not every test, thats why the definition of Stub from previous stub defintion is passed down till it is explicitly defined again in a test.